CalPatRec/src/PhiClusterFinder_module.cc

```diff
@@ -0,0 +1,566 @@
+        fhicl::Atom<float>                                  tMax{                    Name("tMax"),                 Comment("Time histogram end") };
+        fhicl::Atom<float>                                  tBin{                    Name("tBin"),                 Comment("Time histogram bin width") };
+        fhicl::Table<TrkTimeCalculator::Config>             t0Calc{                  Name("t0Calc"),               Comment("TimeTracker calculator config") };
+        fhicl::Atom<float>                                  yMin{                    Name("yMin"),                 Comment("Minimum hit in time hist bin for peak") };
+        fhicl::Atom<float>                                  maxDt{                   Name("maxDt"),                Comment("Maximum time difference between T0 and hit time") };
+        fhicl::Atom<float>                                  minSigma{                Name("minSigma"),             Comment("Min sigma cut to remove compton particles") };
+        fhicl::Atom<int>                                    peakWidth{               Name("peakWidth"),            Comment("Time Peak Width") };
+        fhicl::Atom<art::InputTag>                          ComboHitCollection{      Name("ComboHitCollection"),   Comment("ComboHit collection name") };
+        fhicl::Atom<art::InputTag>                          TimeClusterCollection{   Name("TimeClusterCollection"),Comment("TimeCluster collection name") };
+        fhicl::Atom<bool>                                   useCC{                   Name("useCC"),                Comment("Use calo cluster") };
+        fhicl::Table<PhiClusterFinderTypes::Config>         DiagPlugin{              Name("DiagPlugin"),           Comment("Diag plugin") };
+      };
+
+      explicit PhiClusterFinder(const art::EDProducer::Table<Config>& config);
+      void beginJob();
+      void produce(art::Event& event);
+
+  private:
+    typedef std::pair<Float_t,int> BinContent;
+    int                                 _iev;
+    int                                 _diag,_debug;
+    int                                 _minnsh;
+    int                                 _threshold;
+    int                                 _nphiclusters;
+    float                               _phimin,_phimax;
+    int                                 _nphibins;
+    float                               _mindphi,_pitch;
+    float                               _tmin, _tmax, _tbin;
+    TrkTimeCalculator                   _t0calc;
+    float                               _ymin,_maxdt,_minsigma;
+    int                                 _npeak;
+    art::ProductToken<ComboHitCollection> const _chToken;
+    art::ProductToken<TimeClusterCollection> const _tcToken;
+    bool                                _useCC;
+    const ComboHitCollection*           _chcol;
+    TH1F*                               _hist1;
```

Comment:
For this PR it is OK for me to leave around these TH1F, BUT in a future PR you should switch to `std::array` or `std::vector`.